clang/test/Driver/sycl-bc-device-libraries.cpp

```diff
@@ -0,0 +1,61 @@
+/// Test that SYCL bitcode device libraries are properly separated for NVIDIA and AMD targets
+
+/// Check devicelib and libspirv are linked for nvptx
+// RUN: %clang -### -fsycl --offload-new-driver \
+// RUN:   -fsycl-targets=nvptx64-nvidia-cuda \
+// RUN:   --cuda-path=%S/Inputs/CUDA_102/usr/local/cuda \
+// RUN:   %s 2>&1 | FileCheck -check-prefix=CHECK-NVPTX-BC %s
+
+// CHECK-NVPTX-BC: clang-linker-wrapper
+// CHECK-NVPTX-BC-SAME: "--bitcode-library=nvptx64-nvidia-cuda={{.*}}devicelib-nvptx64-nvidia-cuda.bc" "--bitcode-library=nvptx64-nvidia-cuda={{.*}}libspirv-nvptx64-nvidia-cuda.bc"
+
+/// Check devicelib is linked for amdgcn
+// RUN: %clang -### -fsycl --offload-new-driver \
+// RUN:   -fsycl-targets=amdgcn-amd-amdhsa \
+// RUN:   -Xsycl-target-backend=amdgcn-amd-amdhsa --offload-arch=gfx900 \
+// RUN:   --rocm-path=%S/Inputs/rocm \
+// RUN:   %s 2>&1 | FileCheck -check-prefix=CHECK-AMD-BC %s
+
+// CHECK-AMD-BC: clang-linker-wrapper
+// CHECK-AMD-BC-SAME: "--bitcode-library=amdgcn-amd-amdhsa={{.*}}devicelib-amdgcn-amd-amdhsa.bc"
```

Comment:
```suggestion
// CHECK-AMD-BC-SAME: "--bitcode-library=amdgcn-amd-amdhsa={{.*}}devicelib-amdgcn-amd-amdhsa.bc"
```

Just for my understanding: why we check for NVPTX that `--bitcode-library` appears twice, but here we don't?

Reply:
To check that libspirv is added by the driver, see the other comment about nvptx vs amd.